From ec3495b9659238485548a2b209f34c3e2390f431 Mon Sep 17 00:00:00 2001
From: dann frazier <dann.frazier@chainguard.dev>
Date: Tue, 15 Sep 2026 11:33:31 -0600
Subject: [PATCH 05/18] Revert "env: ensure all -v messages are quoted"

This reverts commit d18d3a85be861c7a68d3ae05ec04b5cbf6487959.
NEWS left untouched.
---
 src/env.c          | 27 ++++++++++++---------------
 tests/env/env-S.pl |  2 +-
 tests/env/env.sh   |  2 +-
 3 files changed, 14 insertions(+), 17 deletions(-)

diff --git a/src/env.c b/src/env.c
index 1e2cfb263..b0842ebee 100644
--- a/src/env.c
+++ b/src/env.c
@@ -223,7 +223,7 @@ unset_envvars (void)
 {
   for (idx_t i = 0; i < usvars_used; ++i)
     {
-      devmsg ("unsetting: %s\n", quote (usvars[i]));
+      devmsg ("unsetting: %s\n", usvars[i]);
 
       if (unsetenv (usvars[i]))
         error (EXIT_CANCELED, errno, _("cannot unset %s"),
@@ -269,7 +269,7 @@ unset_envvars_in_vector (struct env_vector *env)
   for (idx_t i = 0; i < usvars_used; ++i)
     {
       char const *name = usvars[i];
-      devmsg ("unsetting: %s\n", quote (name));
+      devmsg ("unsetting: %s\n", name);
 
       if (! *name || strchr (name, '='))
         error (EXIT_CANCELED, EINVAL, _("cannot unset %s"), quoteaf (name));
@@ -289,7 +289,7 @@ unset_envvars_in_vector (struct env_vector *env)
 static void
 set_envvar (char *assignment, char *eq)
 {
-  devmsg ("setting:   %s\n", quote (assignment));
+  devmsg ("setting:   %s\n", assignment);
 
   if (putenv (assignment))
     {
@@ -304,7 +304,7 @@ set_envvar (char *assignment, char *eq)
 static void
 set_envvar_in_vector (struct env_vector *env, char *assignment, char *eq)
 {
-  devmsg ("setting:   %s\n", quote (assignment));
+  devmsg ("setting:   %s\n", assignment);
 
   idx_t name_length = eq - assignment;
   for (idx_t i = 0; i < env->used; ++i)
@@ -415,12 +415,12 @@ set_envvars_from_file (struct env_vector *env, char const *file)
       char *eq = strchr (assignment, '=');
       if (! eq)
         {
-          devmsg ("adding:    %s\n", quote (assignment));
+          devmsg ("adding:   %s\n", assignment);
           merged[merged_count++] = assignment;
         }
       else
         {
-          devmsg ("setting:   %s\n", quote (assignment));
+          devmsg ("setting:   %s\n", assignment);
 
           char **slot = hash_lookup (slot_table, &assignment);
           if (slot)
@@ -1088,12 +1088,9 @@ main (int argc, char **argv)
              incorrect shebang usage with extraneous space, e.g.:
                 #!/usr/bin/env -i command
              In which case argv[1] == "-i command".  */
-          {
-            char c = optc;
-            error (0, 0, _("invalid option -- %s"), quote_mem (&c, 1));
-            error (0, 0, _("use -[v]S to pass options in shebang lines"));
-            usage (EXIT_CANCELED);
-          }
+          error (0, 0, _("invalid option -- '%c'"), optc);
+          error (0, 0, _("use -[v]S to pass options in shebang lines"));
+          usage (EXIT_CANCELED);
 
         case_GETOPT_HELP_CHAR;
         case_GETOPT_VERSION_CHAR (PROGRAM_NAME, AUTHORS);
@@ -1217,7 +1214,7 @@ main (int argc, char **argv)
 
   if (newdir)
     {
-      devmsg ("change dir: %s\n", quote (newdir));
+      devmsg ("change dir: %s\n", quoteaf (newdir));
 
       if (chdir (newdir) != 0)
         error (EXIT_CANCELED, errno, _("cannot change directory to %s"),
@@ -1227,13 +1224,13 @@ main (int argc, char **argv)
   char *program = argv[optind];
   if (argv0)
     {
-      devmsg ("argv0:     %s\n", quote (argv0));
+      devmsg ("argv0:     %s\n", quoteaf (argv0));
       argv[optind] = argv0;
     }
 
   if (dev_debug)
     {
-      devmsg ("executing: %s\n", quote (program));
+      devmsg ("executing: %s\n", program);
       for (int i=optind; i<argc; ++i)
         devmsg ("   arg[%d]= %s\n", i-optind, quote (argv[i]));
     }
diff --git a/tests/env/env-S.pl b/tests/env/env-S.pl
index 7b0912ab7..3fc645cf8 100755
--- a/tests/env/env-S.pl
+++ b/tests/env/env-S.pl
@@ -246,7 +246,7 @@ my @Tests =
             "env: use -[v]S to pass options in shebang lines\n" .
            "Try 'env --help' for more information.\n"}],
      ['err_sp3', q['-v	-S cat -n'], {EXIT=>125}, # embedded tab after -v
-      {ERR=>"env: invalid option -- '\\t'\n" .
+      {ERR=>"env: invalid option -- '\t'\n" .
             "env: use -[v]S to pass options in shebang lines\n" .
            "Try 'env --help' for more information.\n"}],
 
diff --git a/tests/env/env.sh b/tests/env/env.sh
index b00402deb..65b2e7d6f 100755
--- a/tests/env/env.sh
+++ b/tests/env/env.sh
@@ -187,7 +187,7 @@ for arg in 'argv0' ''; do
 env -v -a short --argv0=$arg ./truetrue 2>err || fail=1
 cat <<EOF >err_exp || framework_failure_
 argv0:     '$arg'
-executing: './truetrue'
+executing: ./truetrue
    arg[0]= '$arg'
 EOF
 compare err_exp err || fail=1
-- 
2.53.0

